### PR TITLE
Add Galaxy JWD Python script

### PR DESCRIPTION
**Pull request by sanjaysrikakulam**
**WHAT:**
Adds a new script with the following functionality:
1. Can get you the path of a JWD given a Galaxy job id: Used when one wants to decode/find/go to the JWD for debugging
2. Can delete JWDs of jobs that failed within the last `X` days: For cleaning up JWDs

## How to test the changes?
(Select all options that apply)
- [ ] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [ ] This is a refactoring of components with existing test coverage.
- [x] Instructions for manual testing are as follows:
  1. For the script to run the following ENVs (same as `gxadmin`) are needed 
   1.1 GALAXY_CONFIG_FILE: Path to the galaxy.yml file
   1.2 GALAXY_LOG_DIR: Path to the Galaxy log directory
   1.3 PGDATABASE: Name of the Galaxy database
   1.4 PGUSER: Galaxy database user
   1.5 PGHOST: Galaxy database host
   2. A `~/.pgpass` file (same as `gxadmin`) with the following format
    1.1 `<pg_host>:5432:*:<pg_user>:<pg_password>` 
   4. Script Python dependencies (Both dependencies are part of Galaxy, so nothing to do)
    1.1 `psycopg2`
    1.2 `xml.dom.minidom`
   5. To run:
    1.1 `python galaxy_jwd.py get_jwd 12345678`
    1.2 `python galaxy_jwd.py clean_jwds --dry_run True --days 30` or `python galaxy_jwd.py clean_jwds --dry_run False --days 30`

## License
- [x] I agree to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT).

### CODE REVIEW

File: scripts/galaxy_jwd.py [hunk 1]
```diff
@@ -0,0 +1,347 @@
...
+    galaxy_log_dir = os.environ.get("GALAXY_LOG_DIR").strip()
+    db_name = os.environ.get("PGDATABASE").strip()
+    db_user = os.environ.get("PGUSER").strip()
+    db_host = os.environ.get("PGHOST").strip()
+    db_password = extract_password_from_pgpass(pgpass_file=os.path.expanduser("~/.pgpass"))
+    object_store_conf = get_object_store_conf_path(galaxy_config_file)
+    backends = parse_object_store(object_store_conf)
+
+    # Connect to Galaxy database
+    db = Database(
```
Comment (mvdbeek) [inline scripts/galaxy_jwd.py:97]: can you use sqlalchemy please ?

File: scripts/galaxy_jwd.py [hunk 2]
```diff
@@ -0,0 +1,347 @@
...
+    """Get the path to the object_store_conf.xml file
+
+    Args:
+        galaxy_config_file (str): Path to the galaxy.yml file
+
+    Returns:
+        str: Path to the object_store_conf.xml file
+    """
+    object_store_conf = ""
+    with open(galaxy_config_file, "r") as config:
```
Comment (mvdbeek) [inline scripts/galaxy_jwd.py:184]: The config file is a yaml file, please parse it as such.

File: scripts/galaxy_jwd.py [hunk 3]
```diff
@@ -0,0 +1,347 @@
...
+        for line in config:
+            if line.strip().startswith("object_store_config_file"):
```
Comment (mvdbeek) [inline scripts/galaxy_jwd.py:186]: The object store file doesn't have to be an xml file, there are different ways to parse them.

File: scripts/galaxy_jwd.py [hunk 4]
```diff
@@ -0,0 +1,347 @@
...
+    Returns:
+        str: Path to the JWD
+    """
+    job_id = str(job_id)
+
+    # Check if object_store_id exists in our object store config
+    if metadata[0] not in backends_dict.keys():
+        raise ValueError(f"Object store id '{metadata[0]}' does not exist in the object_store_conf.xml file")
+
+    jwd_path = f"{backends_dict[metadata[0]]}/0{job_id[0:2]}/{job_id[2:5]}/{job_id}"
```
Comment (mvdbeek) [inline scripts/galaxy_jwd.py:234]: That's very brittle, I think you just want to load up the object store itself.

**Review by mvdbeek — COMMENTED** [review]
This is a nice start, but I would recommend using Galaxy's config file parsing 
Galaxy's database session, Galaxy's object store instance and creating this as part of the galaxy-data package. Take a look at https://github.com/galaxyproject/galaxy/blob/54e9768d077cf24c6d8ba06c26fa56547f493080/lib/galaxy/celery/__init__.py#L86 for a reasonably similar setup. If you do this it'd also be straightforward to turn this into a celery cron task and that would address all my comments.

**Comment by sanjaysrikakulam** [comment]
Closing this as the revised updated one part of celery task is [here](https://github.com/galaxyproject/galaxy/pull/15816)

**Pull request closed without merging**